lib/net.js

```diff
@@ -894,6 +906,12 @@ Socket.prototype.connect = function(options, cb) {
   self._connecting = true;
   self.writable = true;
 
+  // Cork then uncork upon connecting.
+  this.cork();
+  this.once('connect', function() {
+    this.uncork();
+  });
```

Comment:
Before they were kept in the `_pendingData`, `_pendingEncoding` things, and the callback wouldn't be called, so the streams would buffer the rest. This still happens after this commit though (on a smaller scale with `_pendingBuffer`) because `.end()` uncorks completely.